src/pages/groupComparison/Survival.tsx

```diff
@@ -26,12 +27,52 @@ export default class Survival extends React.Component<ISurvivalProps, {}> {
             this.props.store.uidToGroup
         ],
         invoke: () => {
+            let orderedActiveGroupUids = _.map(this.props.store._activeGroupsNotOverlapRemoved.result!,group=>group.uid)
             const partition = this.props.store.patientsVennPartition.result!;
+
+            // ascending sort partition bases on number of groups in each parition.
+            // if they are equal then sort based on the give order of groups
+            partition.sort((a, b) => {
+                let a1Keys = _.reduce(a.key, (acc, next, key) => {
+                    if (next === true) {
+                        acc.push(key);
+                    }
+                    return acc;
+                }, [] as string[]);
+                let a2Keys = _.reduce(b.key, (acc, next, key) => {
+                    if (next === true) {
+                        acc.push(key);
+                    }
+                    return acc;
+                }, [] as string[]);
+
+                if (a1Keys.length !== a2Keys.length) {
+                    return a1Keys.length - a2Keys.length;
+                }
+
+                const x1Count = _.reduce(a1Keys, (acc, next) => {
+                    acc += orderedActiveGroupUids.indexOf(next);
```

Comment:
we really should be avoiding using indexOf for lookups. It gives O(n^2) time complexity. a hashmap is the preferred choice. I have a helper method for this called stringListToIndexSet. Let me know if you need further explanation.